### ISSUE TITLE
Requesting PKI metrics

### ISSUE THREAD

**Issue by sfc-gh-zchen** [hashicorp/vault#12131]
**Describe the bug**

**To Reproduce**
We are unable to see `database.*` metrics in the `sys/metrics` endpoint. We were expecting to use these metrics to monitor our PKI Secret Engine. (https://www.vaultproject.io/docs/internals/telemetry#secrets-engines-metrics)

**Expected behavior**
We should be able to see the `database.*` metrics.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.7.3
* Vault CLI Version (retrieve with `vault version`): 1.7.2
* Server Operating System/Architecture:

Vault server configuration file(s):

```hcl
# Paste your Vault config here.
# Be sure to scrub any sensitive values
```

**Additional context**

**Comment by hsimon-hashicorp** [comment]
Hi @sfc-gh-zchen! Are you looking to have PKI metrics added to telemetry? Since PKI is not a database secrets engine, you won't be able to retrieve database metrics for it. Please let me know if PKI metrics are what you're looking for. Thanks! :)

**Comment by sfc-gh-zchen** [comment]
Hi @hsimon-hashicorp , yes, we would like to see PKI metrics especially errors / failures to obtain certificates from PKI. That's why we thought `database.*` could help. Can you advise on if there exist such metrics in the PKI secret engine?

**Comment by cipherboy** [comment]
That's for the request @sfc-gh-zchen! This has been implemented in #13889 which will be in Vault 1.10 (current `main` branch). Please take a look and let us know if this is missing anything from your PoV. :-)
